### PR TITLE
VOOD-182: Prepare webpack to bundle webviews

### PR DESCRIPTION
**Pull request by alexandre-lagadec-ost**
Fixes Issue VOOD-182

## Changes

Prepare webpack to bundle webviews.

### CODE REVIEW

File: src/webview/debuggerPanel.ts [hunk 1]
```diff
@@ -55,22 +55,20 @@ export class DebuggerPanel {
     this.viewPanel.webview.html = this.panelViewProxy.getHtml(this.viewPanel.webview);
 
     this.viewPanel.webview.onDidReceiveMessage(
-      (message: WebviewMessage) => {
+      (message: DebuggerPanelMessage) => {
+        console.debug('DebuggerPanel.openPanel', 'message', message);
```
Comment (GinoCardillo-OST) [inline src/webview/debuggerPanel.ts:59]:
```suggestion
```

File: src/webview/panel-views/PanelViewProxyMessage.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { PanelViewCommand } from './panelViewProxy';
```
Comment (GinoCardillo-OST) [inline src/webview/panel-views/PanelViewProxyMessage.ts:1]: Rename the file to camelCase.

File: src/webview/panel-views/abstractPanelViewProxy.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  }
+
+  getMediaUri(extension: string, fileNameStub: string): Uri {
+    return this.getUri('media', extension, PathHelper.getFileName(fileNameStub, extension));
+  }
+
+  getUri(...pathSegments: string[]): Uri {
+    return this.webview.asWebviewUri(Uri.joinPath(this.extensionUri, ...pathSegments));
+  }
+}
```
Comment (GinoCardillo-OST) [inline src/webview/panel-views/abstractPanelViewProxy.ts:48]: Export this class to a separate file in the `util` directory.

File: src/webview/panel-views/abstractPanelViewProxy.ts [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+class PathHelper {
+  private readonly extensionUri: Uri;
+
+  private static getFileName(fileNameStub: string, extension: string): string {
+    return `${fileNameStub}.${extension}`;
+  }
+
+  constructor(private readonly webview: Webview, { extensionUri }: ExtensionContext) {
+    this.extensionUri = extensionUri;
```
[the remark this answers is not in the document]
Comment (alexandre-lagadec-ost) [reply src/webview/panel-views/abstractPanelViewProxy.ts:31]: Done.

File: src/webview/panel-views/abstractPanelViewProxy.ts (hunk 3 above)
Comment (alexandre-lagadec-ost) [reply src/webview/panel-views/abstractPanelViewProxy.ts:48]: Done. I moved `PathReplacement` as well.

File: src/webview/panel-views/PanelViewProxyMessage.ts (hunk 2 above)
Comment (alexandre-lagadec-ost) [reply src/webview/panel-views/PanelViewProxyMessage.ts:1]: Done.

File: src/webview/panel-views/debuggerPanel.ts [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+      'message',
+      ({ data }: MessageEvent<PanelViewProxyMessage<RenderingAreaData, RenderingAreaOptions, RenderingAreaUpdateData>>) =>
+        this.onPanelViewProxyMessageReceived(data)
+    );
+  }
+
+  private onPanelViewProxyMessageReceived(
+    message: DebuggerPanelMessage | PanelViewProxyMessage<RenderingAreaData, RenderingAreaOptions, RenderingAreaUpdateData>
+  ): void {
+    console.debug('DebuggerPanel.onPanelViewProxyMessageReceived', 'message', message);
```
Comment (alexandre-lagadec-ost) [inline src/webview/panel-views/debuggerPanel.ts:218]:
```suggestion
```

@GinoCardillo-OST Should I remove this line as well?

File: src/webview/debuggerPanel.ts (hunk 1 above)
Comment (alexandre-lagadec-ost) [reply src/webview/debuggerPanel.ts:59]: Done.

File: src/webview/panel-views/debuggerPanel.ts (hunk 5 above)
Comment (GinoCardillo-OST) [reply src/webview/panel-views/debuggerPanel.ts:218]: yes, please

Comment (alexandre-lagadec-ost) [reply src/webview/panel-views/debuggerPanel.ts:218]: Done.

**Review by PascalSchuermannOst — APPROVED** [review]

**Review by GinoCardillo-OST — APPROVED** [review]

**Pull request merged**